mitiq/lre/tests/test_layerwise_folding.py

```diff
@@ -30,36 +32,41 @@
 test_circuit1_with_measurements.append(ops.measure_each(*qreg1))
 
 
-def test_multivariate_layerwise_scaling():
+@pytest.mark.parametrize("circuit_type", SUPPORTED_PROGRAM_TYPES.keys())
+def test_multivariate_layerwise_scaling(circuit_type):
     """Checks if multiple scaled circuits are returned to fit the required
     folding pattern for multivariate extrapolation."""
-    multiple_scaled_circuits = multivariate_layer_scaling(
-        test_circuit1, 2, 2, 3
-    )
+    circuit = convert_from_mitiq(test_circuit1, circuit_type)
+    scaled_circuits = multivariate_layer_scaling(circuit, 2, 2, 3)
 
-    assert len(multiple_scaled_circuits) == 10
-    folding_pattern = [
-        (1, 1, 1),
-        (5, 1, 1),
-        (1, 5, 1),
-        (1, 1, 5),
-        (9, 1, 1),
-        (5, 5, 1),
-        (5, 1, 5),
-        (1, 9, 1),
-        (1, 5, 5),
-        (1, 1, 9),
-    ]
+    assert len(scaled_circuits) == 10
+    assert all(
+        isinstance(circuit, SUPPORTED_PROGRAM_TYPES._python_type(circuit_type))
+        for circuit in scaled_circuits
+    )
 
-    for i, scale_factor_vector in enumerate(folding_pattern):
-        scale_layer1, scale_layer2, scale_layer3 = scale_factor_vector
-        expected_circuit = Circuit(
-            [ops.H.on_each(*qreg1)] * scale_layer1,
-            [ops.CNOT.on(qreg1[0], qreg1[1]), ops.X.on(qreg1[2])]
-            * scale_layer2,
-            [ops.TOFFOLI.on(*qreg1)] * scale_layer3,
-        )
-        assert expected_circuit == multiple_scaled_circuits[i]
+    if circuit_type == "cirq":
```

Comment:
> Say it fails, we will have to check which of the two concerns it is failing at.

Well, pytest does that for us! E.g. if we have `test_foo.py`
```py
def test_foo():
    assert 1 == 2
    assert "hi" in {"one", "two"}
```
and run `python -m pytest test_foo.py` you'll see

```
    def test_foo():
        assert 1 == 1
>       assert "hi" in {"one", "two"}
E       AssertionError: assert 'hi' in {'one', 'two'}

tt.py:3: AssertionError
================================================= short test summary info ==================================================
FAILED tt.py::test_foo - AssertionError: assert 'hi' in {'one', 'two'}
```
If both assertions are made to be false, it will alert you to the first one only.

---

As far as checking return type, I agree it's not ideal, but I'm not sure what else to do that's not **a lot** more complicated. We could make sure the function correctly folds each type of circuit, but that means defining the correct circuit in every `QPROGRAM` which is tedious.

Reply:
You're right! I've split them up in https://github.com/unitaryfund/mitiq/pull/2547/commits/97b02dc36c092523fa4bf4c1200c3a451f973544.